source/ck_uuid_utils.cpp

```diff
@@ -42,14 +112,94 @@ ck_uuid cppkit::ck_uuid_utils::s_to_uuid(const std::string& uuidS)
     return output;
 }
 
+#if defined(IS_IOS) || defined(IS_DARWIN)
+// converts a single hex char to a number (0 - 15)
+static unsigned char _hexDigitToChar(char ch)
+{
+	// 0-9
+	if (ch > 47 && ch < 58)
+		return ch - 48;
+
+	// a-f
+	if (ch > 96 && ch < 103)
+		return ch - 87;
+
+	// A-F
+	if (ch > 64 && ch < 71)
+		return ch - 55;
+
+	return 0;
+}
+
+static bool _isValidHexChar(char ch)
+{
+	// 0-9
+	if (ch > 47 && ch < 58)
+		return true;
+
+	// a-f
+	if (ch > 96 && ch < 103)
+		return true;
+
+	// A-F
+	if (ch > 64 && ch < 71)
+		return true;
+
+	return false;
+}
+
+// converts the two hexadecimal characters to an unsigned char (a byte)
+static unsigned char _hexPairToChar(char a, char b)
+{
+	return _hexDigitToChar(a) * 16 + _hexDigitToChar(b);
+}
+#endif
+
 void cppkit::ck_uuid_utils::s_to_uuid(const string& uuidS, uint8_t* uuid)
 {
+#if defined(IS_IOS) || defined(IS_DARWIN)
+	char charOne = '\0';
+	char charTwo = '\0';
+	bool lookingForFirstChar = true;
+	unsigned nextByte = 0;
+
+	for (const char &ch : uuidS)
+	{
+		if (ch == '-')
```

Comment:
This reminds me... when I started using Rust's [uuid](https://docs.rs/uuid/0.7.4/uuid/) crate, I learned that while I'm used to the hyphenated style (which also seems to be the de facto representation), I guess in some circles there's a "simple" and "urn" style. I guess the URN style is from [RFC 4122](https://tools.ietf.org/html/rfc4122); not sure where "simple" comes from--maybe that's just something the Rust lib maintainer thought was relevant. All this to say, though, that it looks like both "simple" and "hyphenated" versions should work here, but not the URN version. I'm not suggesting we support the URN way, rather just pointing it out as something I learned in case we ever run into some crazy person trying to pass in URN style UUIDs.

Reply:
Cool... I didn't even know there was another format. :) I wonder if libuuid supports it?

Reply:
This makes it look like libuuid expects the hyphenated version: https://linux.die.net/man/3/uuid_parse